docs/guide/introduction.md

```diff
@@ -16,7 +16,84 @@ In short, it recognizes pain points on both sides of the CMS experience and reli
+
+All modules use [the same API](/reference/module-api/). That shared foundation means that you can give your blog post module powerful features, such as custom command line tasks and API routes. It's the same foundation the core team uses to build the CMS, so it is well tested and designed to be as intuitive as possible.
+
+### Setting up a module
+
+Modules are organized in a folder at the root of a project appropriately named `modules`. Each module gets its own directory inside that with an `index.js` file that contains its configurations. So the blog post code would being in the file:
+
+```
+modules/blog-post/index.js
+```
+
+The module code itself would then be defined in an object, assigned to `module.exports`. This may look familiar if you know [CommonJS](https://nodejs.org/api/modules.html#modules_modules_commonjs_modules) patterns.
+
+```javascript
+// modules/blog-post/index.js
+module.exports = {
+  // ...
+}
+```
+
+The final step to this minimalist module example is to tell Apostrophe that it should be turned on, or instantiated. That is done in the main application file, `app.js` in its own `modules` property.
+
+```js
+// app.js
+require('apostrophe')({
+  modules: {
+    'blog-post': {}
+  }
+});
+```
+
+There is more to do to make modules useful, but that is all it takes to get them running.
+
+### Module inheritance
+
+Inheritance is the glue of the module system. Every module, other than the root module, extends another one. This means that your blog post module, which extends the "piece type" module, has a huge set of features you never have to write.
+
+```javascript
+// modules/blog-post/index.js
+module.exports = {
+  extend: '@apostrophecms/piece-type'
+};
+```
+
+More than that, it means you can easily customize that core ["piece type"](/reference/glossary.md#piece) module (`@apostrophecms/piece-type`) in your project. That will not only apply changes to the project's blog post content type, but also every piece type in Apostrophe core.
+
+Configuring a core or installed module is as simple as creating an `index.js` file for the module in your project. For example, we might want to log the title of every piece when it is published. We would then create the file:
+
+```
+modules/@apostrophecms/piece-type/index.js
+```
+
+Then add the event handler:
+
+```javascript
+// modules/@apostrophecms/piece-type/index.js
+module.exports = {
+  handlers(self) {
+    return {
+      afterPublish: {
+        logPublished (req, data) {
+          console.log(`Published ${data.published.title}`);
+        }
+      }
+    };
+  }
+};
+```
+
+Since every piece type extends that module, they all get the benefits of changes to it. And in the project code we only need to include our changes. All the rest of the module's code is untouched.
+
+The rest of the documentation will include many specific, practical examples of these ideas. For now a couple of main takeaways are:
```

Comment:
```suggestion
The rest of the documentation will include many specific, practical examples of these ideas. For now the main takeaways are:
```
couple means "two". There are 3, but really, these are all of them, imo.

Reply:
It means two in some contexts, but generally means an indefinite small number.